homeassistant/components/amcrest/switch.py

```diff
@@ -69,22 +69,22 @@ def available(self) -> bool:
         """Return True if entity is available."""
         return self._api.available
 
-    def turn_on(self, **kwargs: Any) -> None:
+    async def async_turn_on(self, **kwargs: Any) -> None:
         """Turn the switch on."""
-        self._turn_switch(True)
+        await self._turn_switch(True)
 
-    def turn_off(self, **kwargs: Any) -> None:
+    async def async_turn_off(self, **kwargs: Any) -> None:
         """Turn the switch off."""
-        self._turn_switch(False)
+        await self._turn_switch(False)
 
-    def _turn_switch(self, mode: bool) -> None:
+    async def _turn_switch(self, mode: bool) -> None:
         """Set privacy mode."""
         lower_str = str(mode).lower()
-        self._api.command(
+        await self._api.async_command(
             f"configManager.cgi?action=setConfig&LeLensMask[0].Enable={lower_str}"
         )
 
-    def update(self) -> None:
+    async def async_update(self) -> None:
         """Update switch."""
-        io_res = self._api.privacy_config().splitlines()[0].split("=")[1]
+        io_res = (await self._api.async_privacy_config()).splitlines()[0].split("=")[1]
         self._attr_is_on = io_res == "true"
```

Comment:
```
   async def async_on_demand_update(self) -> None:
        """Update state."""
        self.async_schedule_update_ha_state(True)

    async def async_added_to_hass(self) -> None:
        """Subscribe to update signal."""
        self._unsub_dispatcher = async_dispatcher_connect(
            self.hass,
            service_signal(SERVICE_UPDATE, self._signal_name),
            self.async_on_demand_update,
        )

    async def async_will_remove_from_hass(self) -> None:
        """Disconnect from update signal."""
        assert self._unsub_dispatcher is not None
        self._unsub_dispatcher()
```

This block needs to be updated as well.

You can drop `async_on_demand_update` and call `async_write_ha_state` from the dispatcher

you can drop  `async_will_remove_from_hass` and wrap `async_dispatcher_connect` in `self.async_on_remove`